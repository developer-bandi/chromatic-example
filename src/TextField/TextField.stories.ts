import type { Meta, StoryObj } from "@storybook/react";
import TextField from "./TextField";

const meta = {
  title: "Example/TextFiled",
  component: TextField,
} satisfies Meta<typeof TextField>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    title: "테스트",
    onChange: () => {},
  },
};
