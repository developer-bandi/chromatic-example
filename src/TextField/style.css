.container {
  display: flex;
  gap: 10px;
}

.title {
  font-size: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.field {
  width: 200px;
  height: 30px;
  font-size: 17px;
  padding: 0 10px;
}
