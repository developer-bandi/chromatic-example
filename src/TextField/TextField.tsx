import "./style.css";

const TextField = ({ title, onChange }: any) => {
  return (
    <div className="container">
      <label className="title" data-testid="label" htmlFor={title}>
        {title}
      </label>
      <input
        className="field"
        data-testid="input"
        id={title}
        onChange={onChange}
      ></input>
    </div>
  );
};

export default TextField;
