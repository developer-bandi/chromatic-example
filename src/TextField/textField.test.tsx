// label 클릭시 input foucus
// input 입력시 onChange 핸들러 호출

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import TextField from "./TextField";

test("label을 클릭하면 input이 foucus 됩니다", async () => {
  const title = "일련번호";
  const onChange = jest.fn();

  // ARRANGE
  render(<TextField title="일련번호" onChange={onChange} />);

  // ACT
  userEvent.click(screen.getByText("일련번호"));

  // ASSERT
  expect(screen.getByLabelText(title)).toHaveFocus();
});

test("input에 값을 넣은 만큼 onChange 핸들러가 호출 됩니다", async () => {
  const title = "일련번호";
  const onChange = jest.fn();

  // ARRANGE
  const { container } = render(<TextField title={title} onChange={onChange} />);

  // ACT
  userEvent.click(screen.getByText(title));
  userEvent.type(screen.getByLabelText(title), "hello");

  // ASSERT

  expect(container).toMatchSnapshot();
  expect(onChange).toHaveBeenCalledTimes(5);
});

test("스냅샷 테스트", async () => {
  const { container } = render(
    <TextField title={"일련번호"} onChange={jest.fn()} />
  );

  expect(container).toMatchSnapshot();
});

describe("The Login Page", () => {
  beforeEach(() => {
    // reset and seed the database prior to every test
    cy.exec("npm run db:reset && npm run db:seed");

    // seed a user in the DB that we can control from our tests
    // assuming it generates a random password for us
    cy.request("POST", "/test/seed/user", { username: "jane.lane" })
      .its("body")
      .as("currentUser");
  });

  it("sets auth cookie when logging in via form submission", function () {
    // destructuring assignment of the this.currentUser object
    const { username, password } = this.currentUser;

    cy.visit("/login");

    cy.get("input[name=username]").type(username);

    // {enter} causes the form to submit
    cy.get("input[name=password]").type(`${password}{enter}`);

    // we should be redirected to /dashboard
    cy.url().should("include", "/dashboard");

    // our auth cookie should be present
    cy.getCookie("your-session-cookie").should("exist");

    // UI should reflect this user being logged in
    cy.get("h1").should("contain", "jane.lane");
  });
});
