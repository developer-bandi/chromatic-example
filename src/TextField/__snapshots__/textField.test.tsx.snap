// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input에 값을 넣은 만큼 onChange 핸들러가 호출 됩니다 1`] = `
<div>
  <div
    class="container"
  >
    <label
      class="title"
      data-testid="label"
      for="일련번호"
    >
      일련번호
    </label>
    <input
      class="field"
      data-testid="input"
      id="일련번호"
    />
  </div>
</div>
`;

exports[`스냅샷 테스트 1`] = `
<div>
  <div
    class="container"
  >
    <label
      class="title"
      data-testid="label"
      for="일련번호"
    >
      일련번호
    </label>
    <input
      class="field"
      data-testid="input"
      id="일련번호"
    />
  </div>
</div>
`;
